hw/xbox/xid-sdl.c

```diff
@@ -464,6 +466,59 @@ static void usb_xid_class_initfn(ObjectClass *klass, void *data)
     uc->handle_attach  = usb_desc_attach;
 }
 
+static HapticEmulationMode detect_haptic_emulation_mode(SDL_Haptic *haptic) {
+    if (SDL_HapticQuery(haptic) & SDL_HAPTIC_LEFTRIGHT) {
+        return EMU_HAPTIC_LEFT_RIGHT;
+    }
+
+    return EMU_UNSUPPORTED;
+}
+
+static int haptic_init(USBXIDState *s) {
+    if (SDL_HapticOpened(s->device_index)) { return 0; }
+
+    s->sdl_haptic = SDL_HapticOpenFromJoystick(
+        SDL_GameControllerGetJoystick(s->sdl_gamepad)
```

Comment:
nit: call in call is not ideal. A temporary variable (`sdl_joystick` for example) would make this more readable.